pkg/controller/lifecycle/actuator_test.go

```diff
@@ -191,42 +178,35 @@ var _ = Describe("Actuator", func() {
 					allowUntrustedImages,
 					allowInsecureRegistries,
 					k8sVersion,
-					gep19Monitoring,
 				)
 				Expect(err).ToNot(HaveOccurred())
-				Expect(resources).To(HaveLen(7))
-
-				expectedResources := map[string]string{
-					deploymentKey:               expectedSeedDeployment(replicas, namespace, genericKubeconfigName, shootAccessServiceAccountName, image, lakomConfigConfigMapName, serverTLSSecretName, strconv.FormatBool(useOnlyImagePullSecrets), strconv.FormatBool(allowUntrustedImages), strconv.FormatBool(allowInsecureRegistries)),
-					pdbKey:                      expectedSeedPDB(namespace, withUnhealthyPodEvictionPolicy),
-					lakomConfigConfigMapNameKey: expectedSeedConfigMapLakomConfig(namespace, lakomConfigConfigMapName, lakomConfig),
-					serviceKey:                  expectedSeedService(namespace),
-					serviceAccountKey:           expectedSeedServiceAccount(namespace, shootAccessServiceAccountName),
-					vpaKey:                      expectedSeedVPA(namespace),
-				}
+				Expect(resources).To(HaveKey("data.yaml.br"))
+				compressedData := resources["data.yaml.br"]
+				data, err := test.BrotliDecompression(compressedData)
+				Expect(err).NotTo(HaveOccurred())
 
-				if gep19Monitoring {
-					expectedResources[serviceMonitorKey] = expectedSeedServiceMonitor(namespace)
-				} else {
-					expectedResources[configMapKey] = expectedSeedConfigMap(namespace)
-				}
-
-				for key, expectedResource := range expectedResources {
-					resource, ok := resources[key]
-					Expect(ok).To(BeTrue(), key)
+				manifests := strings.Split(string(data), "\n---\n") // Just '---\n' does not work because of the header/footer in the public keys that match the same manifest separator
+				Expect(manifests).To(HaveLen(7))
 
-					strResource := string(resource)
-					Expect(strResource).To(Equal(expectedResource), func() string {
-						return fmt.Sprintf("\nkey=%q\ngenerated resource\n%s\nexpectedResource\n%s\n", key, strResource, expectedResource)
-					})
+				for i := 0; i < len(manifests)-1; i++ { // Re-add the trailing '\n' removed during the split from the separator above
+					manifests[i] += "\n"
 				}
```

Comment:
Isn't
```
for i, _ := range manifests {
		manifests[i] += "\n"
}
```
considered more readable in Go? I've rarely seen the original for loops used for this case.
I would expect that the Go compiler can infer that the '_' variable isn't used and thus not make a copy of the object since it's not necessary.

Reply:
Yes, but the last element of the slice must not get the `\n` suffix. Then it should be

```go
for i := range manifests {
	if i == len(manifests)-1 {
		break
	}
	manifests[i] += "\n"
}
```

In this case I cannot say which is more readable or idiomatic. 
I am fine with both ways, what is your preference?

Reply:
My bad, didn't notice that the loop skips the last element. I'd much rather prefer your original for loop rather than the range in this case 👍

Reply:
Ok, I however decided to change it to 
```go
for i := range manifests { // Re-add the trailing '\n' removed during the split from the separator above
	if i < len(manifests)-1 {
		manifests[i] += "\n"
	}
}
```

It should be now visible that the last element is skipped.

Reply:
I think we kind of overcomplicated the issue. I still prefer your original for loop but I'm find with this too 😃